docs/migration-guides.md

```diff
@@ -0,0 +1,1356 @@
+
+assert(sizeOf<uint64>() === 8);
+assert(sizeOf<boolean>() === 1);
+assert(sizeOf<Uint<512>>() === 64);
+assert(sizeOf<[StaticArray<Bool, 10>, boolean, boolean]>() === 3);
+```
+
+#### Update `abiCall` syntax to use object parameter
+
+The signature of `abiCall` helper changes from
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+import MyContract from './MyContract.algo'
+
+// ... rest of code
+
+arc4.abiCall(MyContract.prototype.myMethod, { ... })
+```
+
+to
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+import MyContract from './MyContract.algo'
+
+// ... rest of code
+
+arc4.abiCall({ method: MyContract.prototype.myMethod, ... })
+```
+
+The new `method` property exists to provide a natural way to specify the `TMethod` generic parameter, but it is optional and alternatively the generic arg can be specified explicitly with
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+import MyContract from './MyContract.algo'
+
+// ... rest of code
+
+arc4.abiCall<typeof MyContract.prototype.myMethod>({ ... })
+```
+
+This form of invocation supports using type only imports `import type { MyContract } from '...'` which allow for circular references versus value imports which cannot be circular.
+
+If, in some unforeseen circumstances, both a type argument and the `method` property are provided to the `abiCall` helper, the type argument takes precedence and the method specified by the type argument is called.
+
+**BEFORE - Algorand TypeScript beta**
+
+```ts
+import { arc4, assert } from '@algorandfoundation/algorand-typescript'
+import Hello, { HelloStubbed } from './Hello.algo'
+
+// ... rest of code
+
+const result2 = arc4.abiCall(Hello.prototype.greet, {
+  appId: 1234,
+  args: ['abi'],
+}).returnValue;
+
+assert(result2 === 'hello abi');
+
+const result3 = arc4.abiCall(HelloStubbed.prototype.greet, {
+  appId: 1234,
+  args: ['stubbed'],
+}).returnValue;
+assert(result3 === 'hello stubbed');
+```
+
+**AFTER - Algorand TypeScript 1.0**
+
+```ts
+import { arc4, assert } from '@algorandfoundation/algorand-typescript'
+import Hello, { HelloStubbed } from './Hello.algo'
+
+// ... rest of code
+
+// provide method property
+const result2 = arc4.abiCall({
+  method: Hello.prototype.greet,
+  appId: app,
+  args: ['abi'],
+}).returnValue;
+
+assert(result2 === 'hello abi');
+
+// or provide type argument, this approach supports using type only imports
+const result3 = arc4.abiCall<typeof HelloStubbed.prototype.greet>({
+  appId: app,
+  args: ['stubbed'],
+}).returnValue;
+assert(result3 === 'hello stubbed');
+```
+
+#### Rename `interpretAsArc4` function to `convertBytes`
+
+The function now accepts an `options` object with `strategy: 'unsafe-cast' | 'validate'` and an optional `prefix` parameter.
+
+**BEFORE - Algorand TypeScript beta**
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+
+// ... rest of code
+
+const x = arc4.interpretAsArc4<Uint<32>>(a);
```

Comment:
minor: in `beta` it should be `const x = arc4.interpretAsArc4<arc4.UintN<32>>(a);`